thirty-five-mm/app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,71 @@
+class CommentsController < ApplicationController
+
+  def new
+    if params[:film_id]
+      @film = Film.find(params[:film_id])
+      @comment = Comment.new
+      render '_film_new_comment.html.erb', locals: { film: @film, comment: @comment }, layout: false
+    else
+      @review = Review.find(params[:review_id])
+      @comment = Comment.new
+      render '_review_new_comment.html.erb', locals: { review: @review, comment: @comment}, layout: false
+    end
+  end
+
+  def create
```

Comment:
This method needs restructuring. You can do something like 
```
if params[:film_id]
  commentable = Film.find(params[:film_id])
else
  commentable Review.find(params[:review_id])
end
```
Then the rest of the method can do
commentable.comments.build  - you don't need these two very similar branches.